### PR TITLE
Return proper number of expressions for nth_value_agg

### LINKED ISSUE #9028: `NthValueAgg` returns 1 physical expression, but needs 2 when being constructed with `create_aggregate_expr`
### Describe the bug

`NthValueAgg` returns one expression here (which is used when encoding a plan to a protobuf): https://github.com/apache/arrow-datafusion/blob/ed2453901949101556c0c89b9acf442873c06ce8/datafusion/physical-expr/src/aggregate/nth_value.rs#L119-L121

But `create_aggregate_expr` expects two input expressions (which is used when decoding a protobuf to a plan): https://github.com/apache/arrow-datafusion/blob/ed2453901949101556c0c89b9acf442873c06ce8/datafusion/physical-expr/src/aggregate/build_in.rs#L375-L396

A workaround could be downcasting and manually adding in an expression on either the encode or decode side, but that does feel like a hack.

### To Reproduce

I've added an additional test case in this pr which hits the issue: https://github.com/apache/arrow-datafusion/pull/9027/files#diff-f5f4ff60a3b2f61388e01e1ebda773642ab5d86b5acee785c902f24e3b9dad70R340-R348

### Expected behavior

I expect that `expression` returns the proper number of expressions needed to recreate an aggregate expression in `create_aggregate_expr`.

### Additional context

I'm trying to make sure all the aggregates can be properly encoded and decoded into protobufs, and hit this issue.

### PR DESCRIPTION
**Pull request by mustafasrepo**
## Which issue does this PR close?

Closes #9028.

## Rationale for this change
`fn expressions` method of the `nth_value_agg` returns missing information. For this reason, `nth_value_agg` cannot be decoded properly with protobuf. For more explanation see #9028. This PR fixes this problem.

### CODE REVIEW

File: datafusion/proto/tests/cases/roundtrip_physical_plan.rs [hunk 1]
```diff
@@ -337,16 +338,16 @@ fn rountrip_aggregate() -> Result<()> {
             DataType::Float64,
         ))],
         // TODO: See <https://github.com/apache/arrow-datafusion/issues/9028>
```
Comment (comphead) [inline datafusion/proto/tests/cases/roundtrip_physical_plan.rs:340]: do we need a TODO?

Comment (scsmithr) [inline datafusion/proto/tests/cases/roundtrip_physical_plan.rs:340]: We can remove this now, I originally added it in to provide context on why this case was commented out.

**Review by scsmithr — APPROVED** [review]
This looks good to me. Thanks for getting that fixed.

File: datafusion/proto/tests/cases/roundtrip_physical_plan.rs (hunk 1 above)
Comment (alamb) [reply datafusion/proto/tests/cases/roundtrip_physical_plan.rs:340]:
```suggestion
```

**Review by alamb — APPROVED** [review]

File: datafusion/proto/tests/cases/roundtrip_physical_plan.rs (hunk 1 above)
Comment (mustafasrepo) [reply datafusion/proto/tests/cases/roundtrip_physical_plan.rs:340]: Missed it. Thanks for catching it

**Pull request merged**